model/task/task.go

```diff
@@ -2990,11 +2990,14 @@ func GetTasksByVersion(versionID string, opts GetTasksByVersionOptions) ([]Task,
 	pipeline := getTasksByVersionPipeline(versionID, opts)
 
 	if len(opts.BaseStatuses) > 0 {
-		pipeline = append(pipeline, bson.M{
-			"$match": bson.M{
-				BaseTaskStatusKey: bson.M{"$in": opts.BaseStatuses},
-			},
-		})
+		// If we are filtering for all statuses it is more performant to not include the status filter
+		if !utility.StringSliceContains(opts.BaseStatuses, "all") {
```

Comment:
Can we make this a constant somewhere?

Reply:
Actually, if the user chooses "all" on the UI can we just not specify any base statuses? (or can that be handled in the resolver?) Since this is pretty UI specific it seems better to handle outside of this function.

Reply:
1. I ended up moving this check into the pipeline and included it in the guard clause here. We shouldn't be applying base status filters if include base tasks is false
https://github.com/evergreen-ci/evergreen/pull/5396/files#diff-4b29d204064065e99ec7762e466f51f0ac8461fcc5f8fc2eeda53d87a5044060R3363-R3375

Reply:
This isn't quite what I meant; "all" is very UI specific (it's not something that the backend would ever pass in) and since this is used in a lot of places I think it makes sense to do set opts.BaseStatuses to nil before ever calling GetTasksByVersion. Minna's review seems worth looking into to address that though.

Reply:
This pipeline stage is used in several places i think it makes sense to include this logic here as a general guard for everywhere `getTasksByVersionPipeline` as opposed to leaving it up to the implementer. AFAICT these queries are only used on the UI so it makes sense to tie these UI concepts to it. The entire query revolves around UI specific concepts such as display statuses, execution task fold ups, build variant display names.

Reply:
Still, if we're going to keep this in the pipeline I'd like it to be more general than assuming globals from the UI. Can we have a list in global of all TaskStatuses and then filter those out of BaseStatuses before checking length?